### PR TITLE
[12.0][ADD] account_debitnote

### PR DESCRIPTION
**Pull request by kongrattapong**
In some situations, user need to add some service charge or some fees on top of what already invoiced.
This module add new feature to "Add Debit Note" from customer invoice / vendor bill

Ready to review.

### CODE REVIEW

File: account_debitnote/models/account_invoice.py
Comment (kittiu) [inline account_debitnote/models/account_invoice.py]:
```suggestion
        string='Debit Notes',
```

File: account_debitnote/models/account_invoice.py
Comment (kittiu) [inline account_debitnote/models/account_invoice.py]: Do we need to allow edtiable on draft?

File: account_debitnote/README.rst
Comment (kittiu) [inline account_debitnote/README.rst:1]: You can remove this file. I will be create automatically once it is merged.

File: account_debitnote/models/account_invoice.py
Comment (kittiu) [inline account_debitnote/models/account_invoice.py:22]:
```suggestion
        help="List all debit notes being created by this invoice",
    )
```

File: account_debitnote/models/account_invoice.py
Comment (kittiu) [inline account_debitnote/models/account_invoice.py]:
```suggestion
```

File: account_debitnote/wizard/account_invoice_debitnote.py
Comment (kittiu) [inline account_debitnote/wizard/account_invoice_debitnote.py]:
```suggestion
                    raise ValidationError(_("""Cannot create debit note for
```

**Comment by kittiu** [comment]
We should not allow create Debit Note from Debit Note. Please hide "Create Debit Note" button, from the debit note itself. Also check the same in code.
Debit Note should be create from Invoice only.

Steps:
- From an Incoie, INV-001, I can create Debit Note, i.e., DINV-001 -> OK
- From Debit Note, i.e., DINV-001, I can create Debit Note, i.e., DINV-002 -> Not OK.

**Comment by kittiu** [comment]
image

File: account_debitnote/README.rst
Comment (kongrattapong) [reply account_debitnote/README.rst:1]: Okay, I'll remove it.

File: account_debitnote/models/account_invoice.py
Comment (kongrattapong) [reply account_debitnote/models/account_invoice.py:22]: roger that.
I will input it in.

File: account_debitnote/models/account_invoice.py
Comment (kongrattapong) [reply account_debitnote/models/account_invoice.py]: roger that.

**Comment by kongrattapong** [comment]
> We should not allow create Debit Note from Debit Note. Please hide "Create Debit Note" button, from the debit note itself. Also check the same in code.
> Debit Note should be create from Invoice only.
> 
> Steps:
> 
> * From an Incoie, INV-001, I can create Debit Note, i.e., DINV-001 -> OK
> * From Debit Note, i.e., DINV-001, I can create Debit Note, i.e., DINV-002 -> Not OK.

Okay,I'll fix it.

**Comment by kongrattapong** [comment]
Customer Invoices (THB) - Odoo - Google Chrome_020

File: account_debitnote/readme/CONFIGURE.rst
Comment (kittiu) [inline account_debitnote/readme/CONFIGURE.rst]:
```suggestion
- Make sure user has access to "Show Full Accounting Features"
```

Comment (kittiu) [reply account_debitnote/readme/CONFIGURE.rst]: Please review the configuration steps.

File: account_debitnote/readme/DESCRIPTION.rst
Comment (kittiu) [inline account_debitnote/readme/DESCRIPTION.rst]:
```suggestion
In some situations, user need to add some service charge or some fees on top of what already invoiced.
This module add new feature to "Create Debit Note" from customer invoice / vendor bill
```

File: account_debitnote/readme/DESCRIPTION.rst
Comment (kittiu) [inline account_debitnote/readme/DESCRIPTION.rst]:
```suggestion
```

File: account_debitnote/readme/DESCRIPTION.rst
Comment (kittiu) [inline account_debitnote/readme/DESCRIPTION.rst]:
```suggestion
```

File: account_debitnote/readme/USAGE.rst
Comment (kittiu) [inline account_debitnote/readme/USAGE.rst]:
```suggestion
    - Create a new invoice and validate.
```

File: account_debitnote/readme/USAGE.rst
Comment (kittiu) [inline account_debitnote/readme/USAGE.rst]:
```suggestion
    - Click on Add Debit Note and fill in reasons. You can also specify an account date.
```

File: account_debitnote/readme/USAGE.rst
Comment (kittiu) [inline account_debitnote/readme/USAGE.rst]:
```suggestion
You can also do the same with Vendor Bill.
```

**Review by kittiu — APPROVED** [review]

**Comment by kittiu** [comment]
Functional and technical reviewed.
@kongrattapong please squash commit.

**Comment by kittiu** [comment]
@OCA/accounting-maintainers

**Comment by kittiu** [comment]
@OCA/accounting-maintainers

**Comment by kittiu** [comment]
cc @ps-tubtim can you help review this?

**Review by ps-tubtim — APPROVED** [review]
Functional and Code review: OK.

**Pull request merged**
